From 13b2b9aef3992ea3de20cc188c2dbc9a63266d0b Mon Sep 17 00:00:00 2001
From: Yulei Zhang <yulei.zhang@intel.com>
Date: Fri, 11 May 2018 10:17:04 +0800
Subject: [PATCH 28/45] drm/i915/gvt: introduce a new VFIO region for vfio
 device migration

New VFIO region type VFIO_REGION_TYPE_DEVICE_STATE is added
to fetch and restore the status of vfio device during the
live migration.

Signed-off-by: Yulei Zhang <yulei.zhang@intel.com>
---
 drivers/gpu/drm/i915/gvt/kvmgt.c |  2 +-
 include/uapi/linux/vfio.h        | 19 ++++++++++++-------
 2 files changed, 13 insertions(+), 8 deletions(-)

diff --git a/drivers/gpu/drm/i915/gvt/kvmgt.c b/drivers/gpu/drm/i915/gvt/kvmgt.c
index 09f289d..52769e6 100644
--- a/drivers/gpu/drm/i915/gvt/kvmgt.c
+++ b/drivers/gpu/drm/i915/gvt/kvmgt.c
@@ -588,7 +588,7 @@ static int kvmgt_init_migration(struct intel_vgpu *vgpu)
 		return -ENOMEM;
 
 	ret = intel_vgpu_register_reg(vgpu,
-			PCI_VENDOR_ID_INTEL | VFIO_REGION_TYPE_PCI_VENDOR_TYPE,
+			VFIO_REGION_TYPE_DEVICE_STATE,
 			VFIO_REGION_SUBTYPE_DEVICE_STATE,
 			&intel_vgpu_regops_device_state, MIGRATION_IMG_MAX_SIZE,
 			VFIO_REGION_INFO_FLAG_READ |
diff --git a/include/uapi/linux/vfio.h b/include/uapi/linux/vfio.h
index e4d015c..b0a1c51 100644
--- a/include/uapi/linux/vfio.h
+++ b/include/uapi/linux/vfio.h
@@ -301,8 +301,19 @@ struct vfio_region_info_cap_type {
 #define VFIO_REGION_SUBTYPE_INTEL_IGD_HOST_CFG		(2)
 #define VFIO_REGION_SUBTYPE_INTEL_IGD_LPC_CFG		(3)
 
+/*
+ * The region type device state is for save or restore the vfio device during
+ * migration.
+ */
+#define VFIO_REGION_TYPE_DEVICE_STATE		(1 << 30)
 /* Mdev sub-type for device state save and restore */
-#define VFIO_REGION_SUBTYPE_DEVICE_STATE	(4)
+#define VFIO_REGION_SUBTYPE_DEVICE_STATE	(1)
+
+/* Offset in region to save device state */
+#define VFIO_DEVICE_STATE_OFFSET	1
+
+#define VFIO_DEVICE_START	0
+#define VFIO_DEVICE_STOP	1
 
 /*
  * The MSIX mappable capability informs that MSIX data of a BAR can be mmapped
@@ -314,12 +325,6 @@ struct vfio_region_info_cap_type {
  */
 #define VFIO_REGION_INFO_CAP_MSIX_MAPPABLE	3
 
-/* Offset in region to save device state */
-#define VFIO_DEVICE_STATE_OFFSET	1
-
-#define VFIO_DEVICE_START	0
-#define VFIO_DEVICE_STOP	1
-
 /**
  * VFIO_DEVICE_GET_IRQ_INFO - _IOWR(VFIO_TYPE, VFIO_BASE + 9,
  *				    struct vfio_irq_info)
-- 
2.7.4

